rpc/context.go

```diff
@@ -103,11 +91,38 @@ func contextAuthEntity(ctx context.Context) (interface{}, error) {
 }
 
 // MustContextAuthEntity returns the authentication entity associated with this context;
-// it panics if there is none set.
+// it panics if there is none set. This value is opaque and therefore should not be inspected
+// beyond equality checks.
 func MustContextAuthEntity(ctx context.Context) interface{} {
 	authEntity, err := contextAuthEntity(ctx)
 	if err != nil {
 		panic(err)
 	}
 	return authEntity
 }
+
+// ContextWithAuthUniqueID attaches a unique ID for an authenticated entity to the given context.
+func ContextWithAuthUniqueID(ctx context.Context, authUniqueID interface{}) context.Context {
```

Comment:
Why cannot this be a string? We assume it's a string on reads?

Reply:
Because I like to copy and paste